x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/RoleDescriptor.java

```diff
@@ -56,6 +59,8 @@ public class RoleDescriptor implements ToXContentObject, Writeable {
     private final String[] clusterPrivileges;
     private final ConfigurableClusterPrivilege[] configurableClusterPrivileges;
     private final IndicesPrivileges[] indicesPrivileges;
+    @Nullable
+    private final RemoteIndicesPrivileges[] remoteIndicesPrivileges;
```

Comment:
I'm good with making it non-null. 

My original intention was to distinguish between empty and null to omit `remote_indices` from `GET _security/role` responses for the cases when users did not explicitly set `remote_indices`. This would avoid the surprise that all roles suddenly show up with a new field `remote_indices` field even if they have nothing to do with remote operations, post upgrade. 

However, the same issue applies to GetUserPrivs (as I discovered in a separate PR) and that one lends itself even less to nullability, so I'm happy to make `remoteIndicesPrivileges` field not nullable and have it show up as empty in the role APIs.

Reply:
> Actually, I will go with omitting remote_indices if empty. 

If this is for toXContent, yes, this was my intention as well.